src/main/java/org/broadinstitute/hellbender/utils/io/BlockCompressedIntervalStream.java

```diff
@@ -0,0 +1,626 @@
+                       final Header header,
+                       final WriteFunc<T> writeFunc,
+                       final int compressionLevel ) {
+            this.path = path.toString();
+            this.dict = header.getDictionary();
+            this.sampleMap = createSampleMap(header.getSampleNames());
+            this.writeFunc = writeFunc;
+            this.os = path.getOutputStream();
+            this.bcos = new BlockCompressedOutputStream(os, (Path)null, compressionLevel);
+            this.dos = new DataOutputStream(bcos);
+            this.lastInterval = null;
+            this.indexEntries = new ArrayList<>();
+            this.firstBlockMember = true;
+            writeHeader(header);
+        }
+
+        @VisibleForTesting
+        public Writer( final String streamSource,
+                final OutputStream os,
+                final Header header,
+                final WriteFunc<T> writeFunc ) {
+            this.path = streamSource;
+            this.dict = header.getDictionary();
+            this.sampleMap = createSampleMap(header.getSampleNames());
+            this.writeFunc = writeFunc;
+            this.os = os;
+            this.bcos = new BlockCompressedOutputStream(os, (Path)null, DEFAULT_COMPRESSION_LEVEL);
+            this.dos = new DataOutputStream(bcos);
+            this.lastInterval = null;
+            this.indexEntries = new ArrayList<>();
+            this.firstBlockMember = true;
+            writeHeader(header);
+        }
+
+        private Map<String, Integer> createSampleMap( final List<String> sampleNames ) {
+            final Map<String, Integer> sampleMap = new HashMap<>(sampleNames.size() * 3 / 2);
+            for ( final String sampleName : sampleNames ) {
+                sampleMap.put(sampleName, sampleMap.size());
+            }
+            return sampleMap;
+        }
+
+        private void writeHeader( final Header header ) {
+            try {
+                writeClassAndVersion(header.getClassName(), header.getVersion());
+                writeSamples(header.getSampleNames());
+                writeDictionary(header.getDictionary());
+                dos.flush();
+            } catch ( final IOException ioe ) {
+                throw new UserException("can't write header to " + path, ioe);
+            }
+        }
+
+        private void writeClassAndVersion( final String className, final String version )
+                throws IOException {
+            dos.writeUTF(className);
+            dos.writeUTF(version);
+        }
+
+        private void writeSamples( final List<String> sampleNames ) throws IOException {
+            dos.writeInt(sampleNames.size());
+            for ( final String sampleName : sampleNames ) {
+                dos.writeUTF(sampleName);
+            }
+        }
+
+        private void writeDictionary( final SAMSequenceDictionary dictionary ) throws IOException {
+            dos.writeInt(dictionary.size());
```

Comment:
When converting an evidence file from text to bci format using `PrintSVEvidence`, `dictionary` is null here, and I am also getting null errors for the sample metadata. Can you add an optional header to the text formats? 

For the current version of the pipeline, we still need to be able to consume a headerless text format with `PrintSVEvidence` (writing to headerless text). For the next version, we will require headers. In other words, we need to have the following conversion functionality:

1. headered text -> headered text, bci
2. unheadered text -> unheadered text
3. bci -> headered text, bci